Build break on Linux with g++ 8.3.0

Per [vcpkg:10407](https://github.com/microsoft/vcpkg/issues/10407), if you use g++-8.3.0 to build cppgraphqlgen, it fails properly resolve these template method overloads. It works fine on g++-7.5.0, which is stock on Ubuntu:
```
/usr/bin/g++-8   -isystem src/../include -I../src/../include -I../src/../PEGTL/include -isystem /home/bill/vcpkg/installed/x64-linux/include -std=gnu++1z -MD -MT src/CMakeFiles/graphqlservice.dir/__/IntrospectionSchema.cpp.o -MF src/CMakeFiles/graphqlservice.dir/__/IntrospectionSchema.cpp.o.d -o src/CMakeFiles/graphqlservice.dir/__/IntrospectionSchema.cpp.o -c IntrospectionSchema.cpp
In file included from src/../include/graphqlservice/IntrospectionSchema.h:9,
                 from ../src/../include/graphqlservice/Introspection.h:9,
                 from IntrospectionSchema.cpp:4:
../src/../include/graphqlservice/GraphQLService.h: In instantiation of ‘graphql::service::ModifiedResult<Type>::convert(typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type, graphql::service::ResolverParams&&) [with graphql::service::TypeModifier Modifier = (graphql::service::TypeModifier)2; graphql::service::TypeModifier ...Other = {}; Type = graphql::introspection::object::Type; std::enable_if_t<(List == Modifier), std::future<graphql::response::Value> > = std::future<graphql::response::Value>; typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type = graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type> > >&&]::<lambda(auto:4&&, graphql::service::ResolverParams&&)> [with auto:4 = graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type> > >]’:
/usr/include/c++/8/type_traits:2246:26:   required by substitution of ‘template<class _Fn, class ... _Args> static std::__result_of_success<decltype (declval<_Fn>()((declval<_Args>)()...)), std::__invoke_other> std::__result_of_other_impl::_S_test(int) [with _Fn = graphql::service::ModifiedResult<Type>::convert(typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type, graphql::service::ResolverParams&&) [with graphql::service::TypeModifier Modifier = (graphql::service::TypeModifier)2; graphql::service::TypeModifier ...Other = {}; Type = graphql::introspection::object::Type; std::enable_if_t<(List == Modifier), std::future<graphql::response::Value> > = std::future<graphql::response::Value>; typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type = graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type> > >&&]::<lambda(auto:4&&, graphql::service::ResolverParams&&)>; _Args = {graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type>, std::allocator<std::shared_ptr<graphql::introspection::object::Type> > > >, graphql::service::ResolverParams}]’
/usr/include/c++/8/type_traits:2257:55:   required from ‘struct std::__result_of_impl<false, false, graphql::service::ModifiedResult<Type>::convert(typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type, graphql::service::ResolverParams&&) [with graphql::service::TypeModifier Modifier = (graphql::service::TypeModifier)2; graphql::service::TypeModifier ...Other = {}; Type = graphql::introspection::object::Type; std::enable_if_t<(List == Modifier), std::future<graphql::response::Value> > = std::future<graphql::response::Value>; typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type = graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type> > >&&]::<lambda(auto:4&&, graphql::service::ResolverParams&&)>, graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type>, std::allocator<std::shared_ptr<graphql::introspection::object::Type> > > >, graphql::service::ResolverParams>’
/usr/include/c++/8/type_traits:2262:12:   required from ‘struct std::__invoke_result<graphql::service::ModifiedResult<Type>::convert(typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type, graphql::service::ResolverParams&&) [with graphql::service::TypeModifier Modifier = (graphql::service::TypeModifier)2; graphql::service::TypeModifier ...Other = {}; Type = graphql::introspection::object::Type; std::enable_if_t<(List == Modifier), std::future<graphql::response::Value> > = std::future<graphql::response::Value>; typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type = graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type> > >&&]::<lambda(auto:4&&, graphql::service::ResolverParams&&)>, graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type>, std::allocator<std::shared_ptr<graphql::introspection::object::Type> > > >, graphql::service::ResolverParams>’
/usr/include/c++/8/type_traits:2275:12:   required from ‘class std::result_of<graphql::service::ModifiedResult<Type>::convert(typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type, graphql::service::ResolverParams&&) [with graphql::service::TypeModifier Modifier = (graphql::service::TypeModifier)2; graphql::service::TypeModifier ...Other = {}; Type = graphql::introspection::object::Type; std::enable_if_t<(List == Modifier), std::future<graphql::response::Value> > = std::future<graphql::response::Value>; typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type = graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type> > >&&]::<lambda(auto:4&&, graphql::service::ResolverParams&&)>(graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type> > >, graphql::service::ResolverParams)>’
/usr/include/c++/8/future:1712:5:   required by substitution of ‘template<class _Fn, class ... _Args> std::future<typename std::result_of<typename std::decay<_Tp>::type(typename std::decay<_Types>::type ...)>::type> std::async(std::launch, _Fn&&, _Args&& ...) [with _Fn = graphql::service::ModifiedResult<Type>::convert(typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type, graphql::service::ResolverParams&&) [with graphql::service::TypeModifier Modifier = (graphql::service::TypeModifier)2; graphql::service::TypeModifier ...Other = {}; Type = graphql::introspection::object::Type; std::enable_if_t<(List == Modifier), std::future<graphql::response::Value> > = std::future<graphql::response::Value>; typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type = graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type> > >&&]::<lambda(auto:4&&, graphql::service::ResolverParams&&)>; _Args = {graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type>, std::allocator<std::shared_ptr<graphql::introspection::object::Type> > > >, graphql::service::ResolverParams}]’
../src/../include/graphqlservice/GraphQLService.h:584:20:   required from ‘static std::enable_if_t<(List == Modifier), std::future<graphql::response::Value> > graphql::service::ModifiedResult<Type>::convert(typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type, graphql::service::ResolverParams&&) [with graphql::service::TypeModifier Modifier = (graphql::service::TypeModifier)2; graphql::service::TypeModifier ...Other = {}; Type = graphql::introspection::object::Type; std::enable_if_t<(List == Modifier), std::future<graphql::response::Value> > = std::future<graphql::response::Value>; typename graphql::service::ModifiedResult<Type>::ResultTraits<Type, Modifier, Other ...>::future_type = graphql::service::FieldResult<std::vector<std::shared_ptr<graphql::introspection::object::Type> > >&&]’
IntrospectionSchema.cpp:137:113:   required from here
../src/../include/graphqlservice/GraphQLService.h:606:20: error: ‘convert’ was not declared in this scope
      children.push(convert<Other...>(std::move(entry), ResolverParams(wrappedParams)));
```

--- a/include/graphqlservice/GraphQLService.h
+++ b/include/graphqlservice/GraphQLService.h
@@ -486,7 +486,7 @@ struct ModifiedResult
 			std::shared_ptr<U>,
 			U>;
 
-		using future_type = typename std::conditional_t<std::is_base_of_v<Object, Type>,
+		using future_type = typename std::conditional_t<std::is_base_of_v<Object, U>,
 			FieldResult<std::shared_ptr<Object>>,
 			FieldResult<type>>&&;
 	};
@@ -542,7 +542,7 @@ struct ModifiedResult
 
 			promise.set_value(std::move(wrappedResult));
 
-			return convert<Other...>(promise.get_future(), std::move(wrappedParams)).get();
+			return ModifiedResult::convert<Other...>(promise.get_future(), std::move(wrappedParams)).get();
 		}, std::move(result), std::move(params));
 	}
 
@@ -572,7 +572,7 @@ struct ModifiedResult
 
 			promise.set_value(std::move(*wrappedResult));
 
-			return convert<Other...>(promise.get_future(), std::move(wrappedParams)).get();
+			return ModifiedResult::convert<Other...>(promise.get_future(), std::move(wrappedParams)).get();
 		}, std::move(result), std::move(params));
 	}
 
@@ -595,7 +595,7 @@ struct ModifiedResult
 				// i.e. std::vector<bool> on many platforms. Copy the values from the std::vector<> rather than moving them.
 				for (auto entry : wrappedResult)
 				{
-					children.push(convert<Other...>(entry, ResolverParams(wrappedParams)));
+					children.push(ModifiedResult::convert<Other...>(entry, ResolverParams(wrappedParams)));
 					++std::get<size_t>(wrappedParams.errorPath.back());
 				}
 			}
@@ -603,7 +603,7 @@ struct ModifiedResult
 			{
 				for (auto& entry : wrappedResult)
 				{
-					children.push(convert<Other...>(std::move(entry), ResolverParams(wrappedParams)));
+					children.push(ModifiedResult::convert<Other...>(std::move(entry), ResolverParams(wrappedParams)));
 					++std::get<size_t>(wrappedParams.errorPath.back());
 				}
 			}

